drivers/nrf24l01/nrf24l01.py

```diff
@@ -98,28 +103,39 @@ def __init__(self, spi, cs, ce, channel=46, payload_size=16):
         self.flush_rx()
         self.flush_tx()
 
+    def init_spi(self, baudrate):
+        try:
+            master = self.spi.MASTER
+        except AttributeError:
+            self.spi.init(baudrate=baudrate, polarity=0, phase=0)
+        else:
+            self.spi.init(master, baudrate=baudrate, polarity=0, phase=0)
+
     def reg_read(self, reg):
         self.cs.low()
-        self.spi.send_recv(reg)
-        buf = self.spi.recv(1)
+        self.spi.read(1, reg)
+        self.spi.readinto(self.buffer)
         self.cs.high()
-        return buf[0]
+        return self.buffer[0]
 
-    def reg_write(self, reg, buf):
+    def reg_write(self, reg, value=None, buffer=None):
```

Comment:
My motivations for the previous comment were 1) not need to make things complex passing keyword args; 2) trying to make this PR as minimal as possible.  If you really think that it's confusing passing either an int or bytearray to the same function then, yes, a separate method would be a better way to handle it.